syntax = "proto3";
package protos.connectors.assets;

import "google/protobuf/wrappers.proto";
import "protos/base.proto";
import "protos/connectors/connector.proto";

message GrafanaTargetMetricPromQlAssetModel {
  message PromqlMetric {
    message QueryVariableValueOptions {
      google.protobuf.StringValue variable = 1;
      repeated string values = 2;
    }
    message QueryLabelVariableMap {
      google.protobuf.StringValue label = 1;
      google.protobuf.StringValue variable = 2;
    }
    google.protobuf.StringValue target_metric_ref_id = 1;
    google.protobuf.StringValue datasource_uid = 2;
    google.protobuf.StringValue expression = 3;
    repeated QueryLabelVariableMap label_variable_map = 4;
    repeated QueryVariableValueOptions variable_values_options = 5;
  }

  message PanelPromqlMap {
    google.protobuf.StringValue panel_id = 1;
    google.protobuf.StringValue panel_title = 2;
    repeated PromqlMetric promql_metrics = 3;
  }

  google.protobuf.StringValue dashboard_id = 1;
  google.protobuf.StringValue dashboard_title = 2;
  google.protobuf.StringValue dashboard_url = 3;
  repeated PanelPromqlMap panel_promql_map = 4;
}

message GrafanaDatasourceAssetModel {
  google.protobuf.UInt64Value datasource_id = 1;
  google.protobuf.StringValue datasource_uid = 2;
  google.protobuf.StringValue datasource_url = 3;
  google.protobuf.StringValue datasource_name = 5;
  google.protobuf.StringValue datasource_type = 6;
  google.protobuf.UInt64Value datasource_orgId = 7;
  google.protobuf.StringValue datasource_access = 8;
  google.protobuf.StringValue datasource_database = 9;
  google.protobuf.BoolValue datasource_readonly = 10;
  google.protobuf.StringValue datasource_typeName = 11;
  google.protobuf.BoolValue datasource_basicAuth = 12;
  google.protobuf.BoolValue datasource_isDefault = 13;
}

message GrafanaTargetMetricPromQlAssetOptions {
  message GrafanaDashboardOptions {
    message GrafanaPanelOptions {
      google.protobuf.StringValue panel_id = 1;
      google.protobuf.StringValue panel_title = 2;
    }
    google.protobuf.StringValue dashboard_id = 1;
    google.protobuf.StringValue dashboard_title = 2;
    google.protobuf.StringValue dashboard_url = 3;
    repeated GrafanaPanelOptions panel_options = 4;
  }
  repeated GrafanaDashboardOptions dashboards = 1;
}

message GrafanaDatasourceAssetOptions {
  message GrafanaDatasourceOptions {
    google.protobuf.UInt64Value datasource_id = 1;
    google.protobuf.StringValue datasource_uid = 2;
    google.protobuf.StringValue datasource_name = 3;
  }
  repeated GrafanaDatasourceOptions prometheus_datasources = 1;
}

message GrafanaAssetModel {
  google.protobuf.UInt64Value id = 1;
  Source connector_type = 2;
  SourceModelType type = 3;
  sfixed64 last_updated = 4;
  oneof asset {
    GrafanaTargetMetricPromQlAssetModel grafana_target_metric_promql = 5;
    GrafanaDatasourceAssetModel grafana_prometheus_datasource = 6;
  }
}

message GrafanaAssets {
  repeated GrafanaAssetModel assets = 1;
}